### ISSUE TITLE
Send from MBHD wallet to another MBHD wallet not seen after switch wallet

### ISSUE THREAD

**Issue by jim618** [bitcoin-solutions/multibit-hd#573]
I just did a send from a soft wallet to a Trezor wallet (in the same instance of MBHD).
The send was correctly seen on the 'sending' wallet but not seen on the 'receiving' wallet until a Repari wallet was done.

Screenshots:

![screen shot 2015-05-24 at 12 34 59](https://cloud.githubusercontent.com/assets/919427/7787700/c09022d8-0211-11e5-848f-1aa905b4fffe.png)

![screen shot 2015-05-24 at 12 35 25](https://cloud.githubusercontent.com/assets/919427/7787704/c94d9e50-0211-11e5-928e-289130340089.png)

**Comment by jim618** [comment]
Then after a repair wallet:

![screen shot 2015-05-24 at 12 39 27](https://cloud.githubusercontent.com/assets/919427/7787706/ef35dca4-0211-11e5-9c0a-5563fe0230b8.png)

**Comment by jim618** [comment]
I have clarified this issue a little.
If I do this I see the zero conf tx:
1) Pay from MBHD wallet A to MBHD wallet B.
2) Close MBHD and restart. 
3) Log into Wallet B - see new tx

If I do this I DO NOT see the new zero conf tx
1) Pay from MBHD wallet A to MBHD wallet B.
2) SWITCH from Wallet A to Wallet B

**Comment by jim618** [comment]
I have updated bitcoinj-alice and MBHD so that the mempool is now cleared when the Bitcoin network is restarted. This makes MBHD 'see' any new zero conf tx, even if sent by another MBHD wallet in the same instance (i.e. after a switch).

Pushed to staging repo.

To test it, try various combinations of sending to and from softand hard wallets, switching, closing/opening MBHD etc.

Awaiting review and closing
